diff -rub -x .git pico9918.ORG/src/main.c pico9918/src/main.c
--- pico9918.ORG/src/main.c	2024-10-18 12:33:31.440466900 +0900
+++ pico9918/src/main.c	2024-10-18 16:00:03.499524100 +0900
@@ -64,13 +64,13 @@
   */
 
 #define PCB_MAJOR_VERSION 0
-#define PCB_MINOR_VERSION 4
+#define PCB_MINOR_VERSION 3
 
-#define GPIO_CD7 14
+#define GPIO_CD7 12
 #define GPIO_CSR tmsRead_CSR_PIN  // defined in tms9918.pio
 #define GPIO_CSW tmsWrite_CSW_PIN // defined in tms9918.pio
-#define GPIO_MODE 28
-#define GPIO_INT 22
+#define GPIO_MODE 22
+#define GPIO_INT 26
 
 #if PCB_MAJOR_VERSION != 0
 #error "Time traveller?"
@@ -79,8 +79,8 @@
   // pin-mapping for gromclk and cpuclk changed in PCB v0.4
   // in order to have MODE and MODE1 sequential
 #if PCB_MINOR_VERSION < 4
-#define GPIO_GROMCL 29
-#define GPIO_CPUCL 23
+//#define GPIO_GROMCL 29
+#define GPIO_CPUCL 25
 #else
 #define GPIO_GROMCL 25
 #define GPIO_CPUCL 24
@@ -105,7 +105,7 @@
 #define TMS_PIO pio1
 #define TMS_IRQ PIO1_IRQ_0
 
-bi_decl(bi_1pin_with_name(GPIO_GROMCL, "GROM Clock"));
+//bi_decl(bi_1pin_with_name(GPIO_GROMCL, "GROM Clock"));
 bi_decl(bi_1pin_with_name(GPIO_CPUCL, "CPU Clock"));
 bi_decl(bi_pin_mask_with_names(GPIO_CD_MASK, "CPU Data (CD7 - CD0)"));
 bi_decl(bi_1pin_with_name(GPIO_CSR, "Read"));
@@ -187,8 +187,8 @@
 static inline void enableTmsPioInterrupts()
 {
   __dmb();
-  *((io_rw_32*)(PPB_BASE + M0PLUS_NVIC_ICPR_OFFSET)) = 1u << TMS_IRQ;
-  *((io_rw_32*)(PPB_BASE + M0PLUS_NVIC_ISER_OFFSET)) = 1u << TMS_IRQ;
+  *((io_rw_32*)(PPB_BASE + M33_NVIC_ICPR0_OFFSET)) = 1u << TMS_IRQ;
+  *((io_rw_32*)(PPB_BASE + M33_NVIC_ISER0_OFFSET)) = 1u << TMS_IRQ;
 }
 
 /*
@@ -196,7 +196,7 @@
  */
 static inline void disableTmsPioInterrupts()
 {
-  *((io_rw_32*)(PPB_BASE + M0PLUS_NVIC_ICER_OFFSET)) = 1u << TMS_IRQ;
+  *((io_rw_32*)(PPB_BASE + M33_NVIC_ICER0_OFFSET)) = 1u << TMS_IRQ;
   __dmb();
 }
 
@@ -394,7 +394,7 @@
   sm_config_set_out_pins(&readConfig, GPIO_CD7, 8);
   sm_config_set_in_shift(&readConfig, false, false, 32); // L shift
   sm_config_set_out_shift(&readConfig, true, false, 32); // R shift
-  sm_config_set_clkdiv(&readConfig, 1.0f);
+  sm_config_set_clkdiv(&readConfig, 4.0f);
 
   pio_sm_init(TMS_PIO, tmsReadSm, tmsReadProgram, &readConfig);
   pio_sm_set_enabled(TMS_PIO, tmsReadSm, true);
@@ -417,7 +417,7 @@
   tmsPioInit();
 
   // set up the GROMCLK and CPUCLK signals
-  initClock(GPIO_GROMCL, TMS_CRYSTAL_FREQ_HZ / 24.0f);
+  //initClock(GPIO_GROMCL, TMS_CRYSTAL_FREQ_HZ / 24.0f);
   initClock(GPIO_CPUCL, TMS_CRYSTAL_FREQ_HZ / 3.0f);
 
   // wait until everything else is ready, then run the vga loop
diff -rub -x .git pico9918.ORG/src/tms9918.pio pico9918/src/tms9918.pio
--- pico9918.ORG/src/tms9918.pio	2024-10-18 12:33:31.440466900 +0900
+++ pico9918/src/tms9918.pio	2024-10-18 16:00:03.499859200 +0900
@@ -27,7 +27,7 @@
 ;
 
 .program tmsRead
-.define public CSR_PIN 26
+.define public CSR_PIN 20
 
   pull block
   mov x, osr        ; ensure we have a valid fifio value in x
@@ -70,7 +70,7 @@
 ;              | | |e| |    |        |
 
 .program tmsWrite
-.define public CSW_PIN 27
+.define public CSW_PIN 21
 
   wait 0 gpio CSW_PIN [12]   ; wait for CSW to go active (low)
   in pins, 16                ; grab the data CD0-7 and auto-push
diff -rub -x .git pico9918.ORG/src/vga/vga.c pico9918/src/vga/vga.c
--- pico9918.ORG/src/vga/vga.c	2024-10-18 12:33:31.456089800 +0900
+++ pico9918/src/vga/vga.c	2024-10-18 16:00:03.500194100 +0900
@@ -50,10 +50,10 @@
 }
 
 
-#define SYNC_PINS_START 0        // first sync pin gpio number
+#define SYNC_PINS_START 27       // first sync pin gpio number
 #define SYNC_PINS_COUNT 2        // number of sync pins (h and v)
 
-#define RGB_PINS_START  2        // first rgb pin gpio number
+#define RGB_PINS_START  0        // first rgb pin gpio number
 #define RGB_PINS_COUNT 12        // number of rgb pins
 
 #define VGA_PIO         pio0_hw  // which pio are we using for vga?
